src/pages/transactions/[id].js

```diff
@@ -1,14 +1,24 @@
 import { useRouter } from "next/router";
+import { useState } from "react";
 import transactions from '../../data/transactions.json';
 import Link from "next/link";
 import ChevronLeft from "@/src/assets/icons/chevron_left.svg";
+import DeleteBin from "src/assets/icons/delete.svg"
 
-export default function TransactionDetails() {
+export default function TransactionDetails({handleDelete}) {
   const router = useRouter();
   const { id } = router.query;
+  const [modal, setModal] = useState(false);
+  const [deleteId, setDeleteId] = useState(null);
+
   // Find transaction by its ID
   const transaction = transactions.find((transaction) => transaction.id === id);
 
+  const onHandleDelete = (id) => {
```

Comment:
Avoid the state my just passing the (transaction.id) while clicking the delete button and pass this "onClick" as prop to TransactionDetails. I would recommend to stick to naming convention "handleDelete" and "onDelete".

```
function handleDelete() {
onDelete(transaction.id);
}
```